app/animals/otherAnimalController.ts

```diff
@@ -0,0 +1,43 @@
+﻿namespace dogsrus.virtdog {
+  export class OtherAnimalController {
+    public otherAnimals: IAnimal[] = [];
+    public selectedAnimal: IAnimal;
+    public lastAction: string = '';
+
+    static $inject = ['$rootScope', 'eventNames'];
+    constructor(private $rootScope: ng.IRootScopeService, private eventNames: EventNames) {
+      this.intializeAnimalList(eventNames);
+    }
+
+    public performAction() {
+      // in future may keep track of last action so we can respond back and forth
+      this.lastAction = (<any>this.selectedAnimal).defaultAction;
+      this.$rootScope.$broadcast((<any>this.selectedAnimal).defaultAction, this.selectedAnimal);
+    }
+
+    private intializeAnimalList(eventNames: EventNames) {
+      this.otherAnimals.push(
+        {
+          defaultAction: eventNames.catHiss,
+          familiarName: 'Alley Cat',
+          speciesName: ''
+        });
+      this.otherAnimals.push(
+        {
+          defaultAction: eventNames.decapitate,
+          familiarName: 'No Ordinary Rabbit',
+          speciesName: ''
+        });
+      this.otherAnimals.push(
+        {
+          defaultAction: eventNames.dogBark,
+          familiarName: 'Little Noisy Dog',
+          speciesName: ''
+        });
+      this.performAction = this.performAction.bind(this);
+    }
+  }
+  (() => {
```

Comment:
I don't normally put this kind of thing in an iife since it's already encapsulated within a name space. Since this is the same namespace as the module definition, you actually don't need to specify dogsrus.virtdog here